### PR TITLE
add a print_state (and print_state_n) method

### PR DESCRIPTION
**Pull request by zingale**
Useful for dumping out the state for a single zone, for debugging

### CODE REVIEW

**Comment by zingale** [comment]
well, it only outputs a single zone, so it should not be an issue

**Comment by bcfriesen** [comment]
Oh whoops, right you are. I can't read good.

**Comment by zingale** [comment]
this is a prettier version, all in C++ now

File: Src/Base/AMReX_MultiFabUtil.cpp [hunk 1]
```diff
@@ -377,4 +379,34 @@ namespace amrex
...
+
+
+#ifdef _OPENMP
+#pragma omp parallel
+#endif
+      for (MFIter mfi(mf); mfi.isValid(); ++mfi)
+	{
+	  if (mfi.validbox().contains(cell)) {
+	    if (n >= 0) {
+	      amrex::AllPrint() << mf[mfi](cell, n) << std::endl;
```
Comment (WeiqunZhang) [inline Src/Base/AMReX_MultiFabUtil.cpp]: Let's set precision to maximum.  `amrex::AllPrint().SetPrecision(17)`.

**Pull request merged**